// potential updated to gene information retrieval from ncbi, after a bit of back and forth with Google Gemini to parse better
// note I have no implemented this as of yet in hexamadness

// https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=snp&id=6025&report=flt


function getGeneInfo(snpId) {
  const geneURL = `https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=snp&id=${snpId}&report=flt`;

  fetch(geneURL)
    .then(response => response.text())
    .then(xmlString => {
      const parser = new DOMParser();
      const xmlDoc = parser.parseFromString(xmlString, "application/xml");
      const locElement = xmlDoc.querySelector("LOC");
      const clinsigElement = xmlDoc.querySelector("CLINSIG");

      if (locElement && clinsigElement) {
        const geneString = locElement.textContent.trim();
        const clinsig = clinsigElement.textContent.trim();

        $("#rs").prepend(` &nbsp;&nbsp;|&nbsp;&nbsp; ${clinsig}`);
        $("#rs").prepend(` &nbsp;&nbsp;|&nbsp;&nbsp; <em>${geneString}</em> Gene &nbsp;`);
      } else {
        console.error("Gene or ClinSig info not found in XML.");
      }
    })
    .catch(error => {
      console.error("Error fetching or parsing data:", error);
    });
}

// Example usage:
getGeneInfo("6025");

// OUTPUT
// <DocumentSummary uid="6025"><SNP_ID>6025</SNP_ID><ALLELE_ORIGIN/><GLOBAL_MAFS><MAF><STUDY>1000Genomes</STUDY><FREQ>T=0.005934/38</FREQ></MAF><MAF><STUDY>GnomAD</STUDY><FREQ>T=0.017619/2471</FREQ></MAF><MAF><STUDY>HapMap</STUDY><FREQ>T=0.008818/10</FREQ></MAF><MAF><STUDY>KOREAN</STUDY><FREQ>T=0./0</FREQ></MAF><MAF><STUDY>PAGE_STUDY</STUDY><FREQ>T=0.006175/486</FREQ></MAF><MAF><STUDY>TOPMED</STUDY><FREQ>T=0.016744/4432</FREQ></MAF><MAF><STUDY>ALFA</STUDY><FREQ>T=0.022585/3828</FREQ></MAF></GLOBAL_MAFS><GLOBAL_POPULATION/><GLOBAL_SAMPLESIZE>0</GLOBAL_SAMPLESIZE><SUSPECTED/><CLINICAL_SIGNIFICANCE>benign,conflicting-interpretations-of-pathogenicity,uncertain-significance,pathogenic,risk-factor</CLINICAL_SIGNIFICANCE><GENES><GENE_E><NAME>F5</NAME><GENE_ID>2153</GENE_ID></GENE_E></GENES><ACC>NC_000001.11</ACC><CHR>1</CHR><HANDLE>PGA-UW-FHCRC,CLINSEQ_SNP,ACPOP,SGDP_PRJ,EVA_UK10K_TWINSUK,EVA_DECODE,GRF,SNP500CANCER,OMUKHERJEE_ADBS,OMICIA,SC_SNP,HUMANGENOME_JCVI,PAGE_STUDY,EGCUT_WGS,SEATTLESEQ,WIAF-CSNP,GENOMED,EXOME_CHIP,TSC-CSHL,TISHKOFF,JJLAB,BCM-HGSC-SUB,EVA,PAGE_CC,BL,1000G_HIGH_COVERAGE,PJP,EVA_FINRISK,PERLEGEN,PACBIO,DDI,HUGCELL_USP,GMI,GNOMAD,AFFY,NHLBI-ESP,JMKIDD_LAB,OMIM-CURATED-RECORDS,BGI,SANFORD_IMAGENETICS,SSAHASNP,FSA-LAB,SWEGEN,TOPMED,SYSTEMSBIOZJU,EVA_GENOME_DK,1000GENOMES,SSMP,EVA_UK10K_ALSPAC,WI_SSAHASNP,TOMMO_GENOMICS,HUMAN_LONGEVITY,CSHL,BUSHMAN,EVA_SVP,ENSEMBL,CSHL-HAPMAP,WUGSC_SSAHASNP,WEILL_CORNELL_DGM,HAMMER_LAB,USC_VALOUEV,KRGDB,COMPLETE_GENOMICS,ILLUMINA,BCMHGSC_JDW,EVA-GONL,ILLUMINA-UK,EVA_MGP,HGSV,EVA_EXAC</HANDLE><SPDI>NC_000001.11:169549810:C:A,NC_000001.11:169549810:C:T</SPDI><FXN_CLASS>missense_variant,coding_sequence_variant</FXN_CLASS><VALIDATED>by-frequency,by-alfa,by-cluster</VALIDATED><DOCSUM>HGVS=NC_000001.11:g.169549811C&gt;A,NC_000001.11:g.169549811C&gt;T,NC_000001.10:g.169519049T&gt;C,NC_000001.10:g.169519049T&gt;A,NG_011806.1:g.41721G&gt;T,NG_011806.1:g.41721G&gt;A,NM_000130.5:c.1601G&gt;T,NM_000130.5:c.1601G&gt;A,NM_000130.4:c.1601G&gt;T,NM_000130.4:c.1601G&gt;A,NP_000121.2:p.Arg534Leu,NP_000121.2:p.Arg534Gln|SEQ=[C/A/T]|LEN=1|GENE=F5:2153</DOCSUM><TAX_ID>9606</TAX_ID><ORIG_BUILD>52</ORIG_BUILD><UPD_BUILD>156</UPD_BUILD><CREATEDATE>2000/09/19 17:02</CREATEDATE><UPDATEDATE>2022/10/12 17:30</UPDATEDATE><SS>7639,3388517,6408627,12674255,14458528,15384064,17374119,20553759,68784225,74903136,77657762,80768688,84523906,87863257,97982940,105435572,106609833,108686946,111314932,119081774,131709017,138101430,155905226,159699610,160758444,164374094,165406825,167258185,169613390,173964819,181341854,199221578,205068124,218708130,230774116,238413632,244304161,252841304,253592200,276094736,284172225,290659975,342010061,482137838,482205031,483027973,489776588,491305652,491609172,535048159,554827710,648487054,780124263,781957563,835607295,974438194,975794409,1067428640,1068350648,1293348588,1397263621,1426011906,1574463850,1584012785,1585179911,1601439505,1644433538,1685903694,1710931971,1712380140,1795244562,1919031046,1958330594,1966894266,2020031327,2094785025,2094982019,2148058720,2167388222,2624524921,2632590763,2632590764,2632590765,2698036876,2732042355,2746493956,2762504184,2984884728,2985529897,2987924052,3021149801,3343761112,3625553663,3626236235,3630622670,3636036300,3637794640,3641622946,3642803170,3646248642,3651486543,3651486544,3653654022,3655867197,3687989788,3725077992,3727579298,3746938022,3770850803,3783597415,3789224010,3794096072,3823683669,3825580202,3826480620,3836636456,3850224915,3895460800,3983953191,3983953192,3984466849,3984832818,3986013958,3986146521,4016944578,4471980427,5146959541,5237633408,5244658671,5314659755,5322718682,5445207742,5512473836,5518170280,5623917242,5623999742,5624221332,5626802100,5800047216,5800087238,5832811813,5847169028,5847562184,5848277537,5910665153,5936512939,5938603862,5979294376,5979997095,5981198647</SS><ALLELE>H</ALLELE><SNP_CLASS>snv</SNP_CLASS><CHRPOS>1:169549811</CHRPOS><CHRPOS_PREV_ASSM>1:169519049</CHRPOS_PREV_ASSM><TEXT/><SNP_ID_SORT>0000006025</SNP_ID_SORT><CLINICAL_SORT>1</CLINICAL_SORT><CITED_SORT/><CHRPOS_SORT>0169549811</CHRPOS_SORT><MERGED_SORT>0</MERGED_SORT></DocumentSummary>
